pkg/kubelet/kubelet.go

```diff
@@ -875,7 +883,25 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 		kubeDeps.Recorder,
 		volumepathhandler.NewBlockVolumePathHandler())
 
-	klet.backOff = flowcontrol.NewBackOff(backOffPeriod, MaxContainerBackOff)
+	max := MaxContainerBackOff
+	base := containerBackOffPeriod
+	if utilfeature.DefaultFeatureGate.Enabled(features.EnableKubeletCrashLoopBackoffMax) {
+		if kubeCfg.CrashLoopBackOff.MaxSeconds != nil {
+			max = time.Duration(*kubeCfg.CrashLoopBackOff.MaxSeconds)
+			if max < containerBackOffPeriod {
+				base = max
```

Comment:
If `crashloopbackoff.maxSeconds` is set to less than 10 seconds, both `base` (initial) and `max` will be set to the same value under 10 seconds. Does this mean the backoff will occur at a fixed interval?

Reply:
Yes, such a configuration basically means there is no room for growth from the base value. Based on the other validation, the fastest allowable delay will be every 1 second. (Note that technically restarts will still _wall time_ take 10s+ because of the pod worker loop, but the backoff logic won't be the one to introduce any more delay).

Reply:
Thank you for the explanation! Will this behavior be included in the official documentation? It already states that exponential backoff starts at 10 seconds, so some users might be able to anticipate this behavior. However, there may still be users who find it confusing.
https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#restart-policy

Reply:
Yes, this behavior will be included in the official documentation. I haven't started yet but the docs PR is at https://github.com/kubernetes/website/pull/48499. I've added in a commit there starting to flag places, like this, to change.